realm/realm-library/src/objectServer/java/io/realm/mongodb/RemoteMongoClient.java

```diff
@@ -26,8 +28,10 @@
 public class RemoteMongoClient {
 
     private OsRemoteMongoClient osRemoteMongoClient;
+    private CodecRegistry codecRegistry;
 
-    public RemoteMongoClient(RealmUser realmUser, String serviceName) {
+    public RemoteMongoClient(final RealmUser realmUser, final String serviceName, final CodecRegistry codecRegistry) {
```

Comment:
If this constructor is going to be part of the public API we probably need to consider to make the CodecRegistry optional?

But didn't we talk about making the class abstract so we can hide the constructors?

Reply:
Let's keep the constructors public for now and revisit this in the future as agreed.

Reply:
Regarding the codec registry, I think I would like to wait until the functions PR is merged. I think we might have to use a codec registry factory instead of (if I remember it right) having it as a static member inside one of the configuration classes in case we want to give the option of defaulting to a certain registry in here.